sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_base.py

```diff
@@ -0,0 +1,213 @@
+        :param uamqp.Message or pyamqp.Message message: Message to get encoded size of.
+        :rtype: int
+        """
+
+    @abstractmethod
+    def get_remote_max_message_size(self, handler):
+        """
+        Returns max peer message size.
+        :param AMQPClient handler: Client to get remote max message size on link from.
+        :rtype: int
+        """
+
+    @abstractmethod
+    def create_retry_policy(self, config):
+        """
+        Creates the error retry policy.
+        :param ~azure.eventhub._configuration.Configuration config: Configuration.
+        """
+
+    @abstractmethod
+    def create_link_properties(self, link_properties):
+        """
+        Creates and returns the link properties.
+        :param dict[bytes, int] link_properties: The dict of symbols and corresponding values.
+        :rtype: dict
+        """
+
+    @abstractmethod
+    def create_send_client(self, *, config, **kwargs):
+        """
+        Creates and returns the send client.
+        :param ~azure.eventhub._configuration.Configuration config: The configuration.
+
+        :keyword str target: Required. The target.
+        :keyword JWTTokenAuth auth: Required.
+        :keyword int idle_timeout: Required.
+        :keyword network_trace: Required.
+        :keyword retry_policy: Required.
+        :keyword keep_alive_interval: Required.
+        :keyword str client_name: Required.
+        :keyword dict link_properties: Required.
+        :keyword properties: Required.
+        """
+
+    @abstractmethod
+    def send_messages(self, producer, timeout_time, last_exception, logger):
+        """
+        Handles sending of event data messages.
+        :param ~azure.eventhub._producer.EventHubProducer producer: The producer with handler to send messages.
+        :param int timeout_time: Timeout time.
+        :param last_exception: Exception to raise if message timed out. Only used by uamqp transport.
+        :param logger: Logger.
+        """
+
+    # TODO: delete after data property added to uamqp.BatchMessage
+    #@abstractmethod
+    #def get_batch_message_data(self, batch_message):
+    #    """
+    #    Gets the data body of the BatchMessage.
+    #    :param batch_message: BatchMessage to retrieve data body from.
+    #    """
+
+    @abstractmethod
+    def set_message_partition_key(self, message, partition_key, **kwargs):
+        """Set the partition key as an annotation on a uamqp message.
+
+        :param message: The message to update.
+        :param str partition_key: The partition key value.
+        :rtype: None
+        """
+
+    @abstractmethod
+    def add_batch(self, batch_message, outgoing_event_data, event_data):
+        """
+        Add EventData to the data body of the BatchMessage.
+        :param batch_message: BatchMessage to add data to.
+        :param outgoing_event_data: Transformed EventData for sending.
+        :param event_data: EventData to add to internal batch events. uamqp use only.
```

Comment:
unfortunately, I couldn't find a way to do this in the shared SDK code without doing separate calls like:
if `uamqp_transport`:
   amqp_transport.add_batch(..., event_data)
else:
   amqp_transport.add_batch(...)

which I thought would've been messy, since there are def other places that pass in diff params :(
Thoughts/alternatives?